ui/src/forms/vue-custom-element.ts

```diff
@@ -0,0 +1,481 @@
+// Copied from this PR https://github.com/vuejs/core/pull/4404
+// that adds a `{ shadowRoot: false }` option to `defineCustomElement`.
+// I also force-set the `appContext` on created components so they can
+// access registered and configured Oruga components.
```

Comment:
We need our own copy because of that shadow dom option?

Reply:
Because of:
1. The option to not create a shadow DOM (in order for the app styles to apply to the components)
2. Setting the app context on the components (otherwise, they don't have access to things that are configured on the app itself. Mostly Oruga components and config)

These 2 things basically replicate the behavior we had with `vue-custom-element`. I think [this comment](https://github.com/karol-f/vue-custom-element/issues/259#issuecomment-933008632) summarizes the situation.